GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -405,11 +405,23 @@ protected void ValidatePathParameter(string path)
         {
             if (!repo.CommitAndRootTreeExists(commitId))
             {
-                if (!gitObjects.TryDownloadCommit(commitId))
-                {
-                    error = "Could not download commit " + commitId + " from: " + Uri.EscapeUriString(objectRequestor.CacheServer.ObjectsEndpointUrl);
-                    return false;
-                }
+                return this.ForceTryDownloadCommit(commitId, objectRequestor, gitObjects, out error);
+            }
+
+            error = null;
+            return true;
+        }
+
+        protected bool ForceTryDownloadCommit(
+            string commitId,
+            GitObjectsHttpRequestor objectRequestor,
+            GVFSGitObjects gitObjects,
+            out string error)
+        {
+            if (!gitObjects.TryDownloadCommit(commitId))

```

Comment:
Does the objects endpoint always respond with all of the trees reachable by `commitId`?

Reply:
All trees reachable from its root tree, yes.